app/assets/javascripts/discourse/app/mapping-router.js

```diff
@@ -134,12 +134,21 @@ export function mapRoutes() {
     }
   });
 
-  return BareRouter.extend({
+  const extendedRouter = BareRouter.extend({
     rootURL: getURL("/"),
   }).map(function () {
     tree.mapRoutes(this);
     this.route("unknown", { path: "*path" });
   });
+
+  _extendedRouter = extendedRouter;
+  return extendedRouter;
+}
+
+let _extendedRouter;
+export function unRegisterRouter() {
+  _extendedRouter.dslCallbacks[0] = null;
```

Comment:
I would like a better solution than this, but couldn't find anything, please be my guest!

Reply:
Two notes:
1. I'd rename the function to `teardownRouter` (or just `teardown` if it gets rolled into the Router class) since it doesn't really removes any registration/references to the instance, but modifies/clear its internal state
2. I think `_extendedRouter.dslCallbacks = []` will achieve the same?

Reply:
I tried to do `= []` but it doesn’t work, for a reason I can't explain sadly :(

Reply:
Okay, we'll leave it as an exercise for curious souls. 😉 (maybe me on a weekend 😂)